src/uct/sm/mm/mm_posix.c

```diff
@@ -281,6 +282,19 @@ uct_posix_alloc(uct_md_h md, size_t *length_p, ucs_ternary_value_t hugetlb,
 
     status = UCS_ERR_NO_MEMORY;
 
+    /* immediately unlink the file */
+    ret = (uuid & UCT_MM_POSIX_SHM_OPEN) ? shm_unlink(file_name) : unlink(file_name);
+    if (ret != 0) {
+        ucs_warn("unable to unlink the shared memory segment. File name is: %s",
```

Comment:
@yosefe Do you mean we only need to do unlink() and can delete shm_unlink() at line 286?